cli/tests/testdata/npm/lock_file/lock.json

```diff
@@ -0,0 +1,4 @@
+{
+    "npm:chalk@5.0.1": "sha512-foobar"
```

Comment:
Before landing this, I think we'll have to ensure that npm:chalk@5 always resolves to 5.0.1. Right now, it seems like it could resolve to any version >= 5 < 6.

Reply:
Any pointers how to do that?

Reply:
I think we'll need to store what version each npm package references resolves to, then also do the same for each npm version requirement in each npm dependency.

When loading a lockfile, we can create an `NpmResolutionSnapshot` from that lockfile. It would be great if the lockfile can work in an additive way so that multiple scripts and even running npm package binaries in the repo can all add to the lockfile. Then if the user wants to "reset" it, they could delete the lockfile and re-run everything again.

Reply:
> I think we'll need to store what version each npm package references resolves to, then also do the same for each npm version requirement in each npm dependency.

I'm not so knowledgable about this part. I'll be happy to discuss it offline.

> When loading a lockfile, we can create an NpmResolutionSnapshot from that lockfile. It would be great if the lockfile can work in an additive way so that multiple scripts and even running npm package binaries in the repo can all add to the lockfile. Then if the user wants to "reset" it, they could delete the lockfile and re-run everything again.

I believe that's the current behavior when you pass `--lock-write`. Also I think we could leverage information from lockfile to download all packages upfront instead of going through current discovery phase (which I think could be faster?)